### PR TITLE
[Opt] Extract constants to top-level

### PR DESCRIPTION
**Pull request by xumingkuan**
Related issue = #656 
Benchmark:
**Before**:
benchmark20200428_0

**After**:
benchmark20200428

mpm99.py: 2875 -> 2868 stmts

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: taichi/transforms/compile_to_offloads.cpp [hunk 1]
```diff
@@ -73,6 +73,10 @@ void compile_to_offloads(IRNode *ir,
     irpass::analysis::verify(ir);
   }
 
+  irpass::extract_constant(ir);
```
Comment (yuanming-hu) [inline taichi/transforms/compile_to_offloads.cpp:76]: Quick question: If we extract constants before offloading, does that lead to more usage of `GlobalTmp`? Or we have global temp store forwarding that fixes this?

Comment (xumingkuan) [reply taichi/transforms/compile_to_offloads.cpp:76]: I think #812 fixes this.

Comment (xumingkuan) [reply taichi/transforms/compile_to_offloads.cpp:76]: (Although #812 may introduce more copies of constants...)

This pass reduces the number of statements from 83.059% to 83.052%.

Comment (yuanming-hu) [reply taichi/transforms/compile_to_offloads.cpp:76]: Sounds good :-) Feel free to merge once CI passes.

**Pull request merged**